### PR TITLE
Implemented workspace wipeout

### PR DESCRIPTION
**Pull request by pjanouse**

### CODE REVIEW

**Comment by olivergondza** [comment]
I have a couple of comments:

- What advantage we gain from renaming the fir before (synchronously) deleting it? The effect seems to be the same when it succeeds but it will leave the trash somewhere unexpected when it fails cluttering the disk eventually.
- AFAIU failure to delete the dir does not abort the node termination. Logging the problems to slave log might be suboptimal as it will be removed with the node (cloud statistics does not preserve that) in a short moment for someone to read it. Java logger is likely sufficient.
- In case of `InterruptedException`, it should be enough to just logs it is interrupted and set back the interruption flag: https://stackoverflow.com/a/3976527/2091470

**Comment by pjanouse** [comment]
@olivergondza 
* The advantage of renaming first I can see in the fact we can easy say what can be deleted (later, when the action had failed before) - probably better suffix should be used - like delete_me. If we don't rename the workspace first, we can mix live data with the garbage (in the workspace dir)... - in my opinion it is important to have an option how to say EngOps "Hey, please delete this and that without worrying". If everything works correctly no more diskspace should be used.
* A failure is logged to *both* places - to the server log and to the slave log. I think it is better in this case.
* Agree with the third item, modifying...
* If the Wipeout should be active by default depends, I don't have a strong preference here as it can be enabled/disabled by node XML def anytime, so I'm setting it Enabled by default now

**Comment by olivergondza** [comment]
> The advantage of renaming first I can see in the fact we can easy say what can be deleted

While that is valid, there is still a chance such directories will be accumulated without anyone noticing and the discovery of the true problem (grid fails to delete workspace) is delayed until the disk is full. One more possible caveat with renaming/deleting the workspace including parent directory is the user might not have permissions to do that (think of slave workspace being a mounted filesystem or jenkins user not having write permission in slave remoteFs). The question is if clear indication of what is a leftover thing from the past and can by safely deleted now outweighs these limitations. I have a mild preference in keeping things simple, leaving the dir where it is and deleting the content only.

**Comment by pjanouse** [comment]
Well, I've modified the impl. as requested...

File: plugin/src/main/java/com/redhat/jenkins/nodesharingfrontend/SharedNode.java [hunk 1]
```diff
@@ -45,6 +47,8 @@
         @Override public String getShortDescription() { return "ReservationTasks should not run here"; }
     };
 
+    private boolean wipeout = true;
```
Comment (olivergondza) [inline plugin/src/main/java/com/redhat/jenkins/nodesharingfrontend/SharedNode.java]: The filed initializer here will be used only when instance is created via constructor which it typically is not as we use xstream deserialization to instantiate. This should be set by `readResolve()` or the semantics will needs to be inverted so the default value, `false`, would achieve what we want.

**Pull request merged**